api/v1alpha1/maasvalidator_types.go

```diff
@@ -22,47 +22,96 @@ import (
 func (s MaasValidatorSpec) ResultCount() int {
-	return len(s.MaasInstanceRules.OSImages)
+	return len(s.OSImageRules) + len(s.InternalDNSRules) + len(s.ExternalDNSRules) + len(s.ResourceAvailabilityRules)
 }
 
-// MaasInstance describes the MaaS host
-type MaasInstance struct {
-	// Host is the URL for your MaaS instance
-	Host string `json:"host" yaml:"host"`
-	Auth Auth   `json:"auth" yaml:"auth"`
+// Auth provides authentication information for the MaaS Instance
+type Auth struct {
+	// +kubebuilder:validation:Optional
+	SecretName string `json:"secretName" yaml:"secretName"`
+	// +kubebuilder:validation:Optional
+	TokenKey string `json:"token" yaml:"token"`
 }
 
-// MaasInstanceRules defines the rules for the MaaS instance
-type MaasInstanceRules struct {
-	// Unique rule name
+// OSImageRule defines a rule for validating one or more OS images
+type OSImageRule struct {
+	// Unique name for the rule
 	Name string `json:"name" yaml:"name"`
-	// OSImages is a list of bootable os images
-	OSImages []OSImage `json:"bootable-images,omitempty" yaml:"bootable-images,omitempty"`
-
-	// Auth provides authentication information for the MaaS Instance
-	Auth Auth `json:"auth" yaml:"auth"`
+	// The list of OS images to validate
+	OSImages []OSImage `json:"osImages" yaml:"osImages"`
 }
 
-// OSImage defines an OS Image
+// OSImage defines one OS image
 type OSImage struct {
 	// The name of the bootable image
-	Name string `json:"name" yaml:"name"`
+	OSName string `json:"osName" yaml:"osName"`
 	// OS Architecture
-	Architecture string `json:"os-arch" yaml:"os-arch"`
+	Architecture string `json:"osArch" yaml:"osArch"`
 }
 
-// Auth provides authentication information for the MaaS Instance
-type Auth struct {
-	// +kubebuilder:validation:Optional
-	SecretName string `json:"secretName" yaml:"secretName"`
-	// +kubebuilder:validation:Optional
-	TokenKey string `json:"token" yaml:"token"`
+// InternalDNSRule provides rules for the internal DNS server
+type InternalDNSRule struct {
+	// The domain name for the internal DNS server
+	MaasDomain string `json:"maasDomain" yaml:"maasDomain"`
+	// The DNS records for the internal DNS server
+	DNSRecords []DNSRecord `json:"dnsRecords" yaml:"dnsRecords"`
+}
+
+// DNSRecord provides an internal DNS record
+type DNSRecord struct {
+	// The hostname for the DNS record
+	Hostname string `json:"hostname" yaml:"hostname"`
+	// The IP address for the DNS record
+	IP string `json:"ip" yaml:"ip"`
+	// The type of DNS record
+	Type string `json:"type" yaml:"type"`
+	// Optional Time To Live (TTL) for the DNS record
+	TTL int `json:"ttl,omitempty" yaml:"ttl,omitempty"`
+}
+
+// ExternalDNSRule provides rules for validating the external DNS server
+type ExternalDNSRule struct {
+	// Unique name for the rule
+	Name string `json:"name" yaml:"name"`
+	// Whether the external DNS server is enabled
+	Enabled bool `json:"enabled" yaml:"enabled"`
```

Comment:
Instead of just Enabled, let's do `NumDNSServers`. Then rule will ensure that at least `NumDNSServers` upstream DNS servers are configured.